py/configu/commands/eval_command.py

```diff
@@ -24,90 +23,92 @@ class EvaluatedConfigOrigin(Enum):
     EmptyValue = "EMPTY_VALUE"
 
 
-EvalCommandReturnContext = TypedDict(
-    "EvalCommandReturnContext",
-    {
-        "store": str,
-        "set": str,
-        "schema": str,
-        "key": str,
-        "cfgu": Union[ConfigSchemaContentsValue, Cfgu],
-    },
-)
+class EvalCommandReturnContext(TypedDict):
+    store: str
+    set: str
+    schema: str
+    key: str
+    cfgu: Cfgu
 
-EvalCommandReturnResult = TypedDict(
-    "EvalCommandReturnResult",
-    {
-        "origin": EvaluatedConfigOrigin,
-        "source": str,
-        "value": str,
-    },
-)
 
-EvalCommandReturnValue = TypedDict(
-    "EvalCommandReturnValue",
-    {
-        "context": EvalCommandReturnContext,
-        "result": EvalCommandReturnResult,
-    },
-)
-EvalCommandReturn = Dict[str, EvalCommandReturnValue]
+class EvalCommandReturnResult(TypedDict):
+    origin: EvaluatedConfigOrigin
+    source: str
+    value: str
+
+
+class EvalCommandReturnValue(TypedDict):
+    context: EvalCommandReturnContext
+    result: EvalCommandReturnResult
 
-EvalCommandParameters = TypedDict(
-    "EvalCommandParameters",
-    {
-        "store": ConfigStore,
-        "set": ConfigSet,
-        "schema": ConfigSchema,
-        "configs": Optional[Dict[str, str]],
-        "validate": Optional[bool],
-        "previous": Optional[EvalCommandReturn],
-    },
-    total=False,
-)
+
+class EvalCommandReturn(Dict[str, EvalCommandReturnValue]):
+    """
+    Dict of keys:EvalCommandReturnValue`
+    """
+
+
+class EvalCommandParameters(TypedDict):
+    store: ConfigStore
+    set: ConfigSet
+    schema: ConfigSchema
+    configs: Dict[str, str]
+    validate: bool
+    previous: EvalCommandReturn
 
 
 class EvalCommand(Command[EvalCommandReturn]):
     """
-    The Eval command is used to fetch and validate `Config`s from `ConfigStore`
+    The Eval command is used to fetch and validate `Config`s from ConfigStore
     on demand.
     """
 
     parameters: EvalCommandParameters
 
-    def __init__(self, parameters: EvalCommandParameters) -> None:
+    def __init__(
+        self,
+        *,
+        store: ConfigStore,
+        set: ConfigSet,
+        schema: ConfigSchema,
+        configs: Dict[str, str] = None,
```

Comment:
Why do you think it's not optional anymore? I just removed the typing because it makes no difference.
It's bad practice as far as I know to init mutable params at __init__ (eg `configs={}`)
It wouldn't change the logic in anyway but it keeps the __init__ clean of values, hinting the user that

Reply:
Not optional in that the `Optional` typing was removed (see previous comment). I certainly didn't know that mutable init default params are bad practice...read up a little and found out why. I certainly learned something new today :)

Based on that I'd agree that `None` is the right default value to use here, however in that case I'd suggest changing the typing here to either `Optional[Dict[str,str]]` or `Union[Dict[str,str],None]`, so that type checking doesn't identify this as an error.

Reply:
Superb. Optional is back since prev comment so we're good here.